### ISSUE TITLE
PKCS5 and PKCS12 test cases not executed

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#9580]
`analyze_outcomes.py` reports that the following test cases are never executed:

```
        'test_suite_pkcs12': [
            'PBE Decrypt, (Invalid padding & PKCS7 padding disabled)',
            'PBE Encrypt, pad = 8 (PKCS7 padding disabled)',
        ],
        'test_suite_pkcs5': [
            'PBES2 Decrypt (Invalid padding & PKCS7 padding disabled)',
            'PBES2 Encrypt, pad=6 (PKCS7 padding disabled)',
            'PBES2 Encrypt, pad=8 (PKCS7 padding disabled)',
        ],
```

Fix this.

**Comment by gabor-mezei-arm** [comment]
The `MBEDTLS_CIPHER_PADDING_PKCS7` is a dependency of `MBEDTLS_PKCS5_C` thus it cannot be disabled.

https://github.com/Mbed-TLS/mbedtls/blob/6707584614763ed08ebeaa73866ff28a71dd0538/include/mbedtls/config_adjust_legacy_crypto.h#L301-L307

**Comment by valeriosetti** [comment]
I took a look at this and IMO at least for `development` and `1.1` branches we can simply drop these exceptions. Here's the rationale:
- Support for PKCS12 has been removed from tf-psa-crypto since https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/389.
- PKCS5 is now a private module and the only user is PK which uses it to decrypt encrypted PKCS8 private keys. PK only calls `mbedtls_pkcs5_pbes2_ext` when `MBEDTLS_CIPHER_PADDING_PKCS7` is defined. Also, as Gabor [pointed out above](https://github.com/Mbed-TLS/mbedtls/issues/9580#issuecomment-2428696771), we auto enable `MBEDTLS_CIPHER_PADDING_PKCS7` whenever `MBEDTLS_PK_PARSE_C && MBEDTLS_PKCS5_C` are enabled.

**Comment by gilles-peskine-arm** [comment]
> we can simply drop these exceptions

Er, can we? I thought we'd just drop the tests because it's a useless configuration, and we'd auto-enable `MBEDTLS_CIPHER_PADDING_PKCS7` in the corner case where it doesn't already happen?

**Comment by valeriosetti** [comment]
> > we can simply drop these exceptions
> 
> Er, can we? I thought we'd just drop the tests because it's a useless configuration, and we'd auto-enable `MBEDTLS_CIPHER_PADDING_PKCS7` in the corner case where it doesn't already happen?

Well, yes. Removing the tests happens (or happened, in case of PKCS12) in tf-psa-crypto, but this issue is for Mbed TLS, so here I'm just removing the exceptions. Am I missing something?

**Comment by gilles-peskine-arm** [comment]
Sure, removing the exceptions is the only thing that happens in the Mbed TLS repository. But we still need to remove the test cases in TF-PSA-Crypto first.

**Comment by valeriosetti** [comment]
> Sure, removing the exceptions is the only thing that happens in the Mbed TLS repository. But we still need to remove the test cases in TF-PSA-Crypto first.

Yeah. I was doing the cleanup in tf-psa-crypto in a security PR, but I'm now thinking to move that part in a public PR. Let me reorganize stuff and I'll probably ask for some review soon
